crates/router/src/core/payments/tokenization.rs

```diff
@@ -66,16 +68,16 @@ where
             } else {
                 None
             };
-            let future_usage_validation = resp
+            let customer_acceptance = resp
                 .request
-                .get_setup_future_usage()
-                .map(|future_usage| {
-                    (future_usage == storage_enums::FutureUsage::OffSession && is_mandate)
-                        || (future_usage == storage_enums::FutureUsage::OnSession && !is_mandate)
-                })
-                .unwrap_or(false);
+                .get_customer_acceptance()
+                .encode_to_value()
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Unable to serialize customer acceptance to value")?;
 
-            let pm_id = if future_usage_validation {
+            let pm_id = if resp.request.get_customer_acceptance().is_some()
+                && resp.request.get_setup_future_usage().is_some()
+            {
```

Comment:
Are we sure we don't need the `future_usage_validation` condition here?

Reply:
Yes, we don't now that `customer_acceptance` is common between both of the `setup_future_usages` we don't want to say, that save the card only when `off_session+customer_acceptance` OR `on_session`.

Reply:
right